autoload/autopairs/Keybinds.vim

```diff
@@ -1,5 +1,30 @@
 " Always silent the command
-inoremap <silent> <SID>AutoPairsReturn <C-R>=autopairs#AutoPairsReturn()<CR>
+function! autopairs#Keybinds#IgnoreInsertEnter(f) abort
+    let l:pre = "\<C-r>=autopairs#Keybinds#SetEventignore()\<CR>"
+    let l:val = call(function(a:f), a:000)
+    let l:post = "\<C-r>=autopairs#Keybinds#ResetEventignore()\<CR>"
+    return l:pre . l:val . l:post
+endfunction
+
+function! autopairs#Keybinds#IgnoreInsertEnterCmd(cmd) abort
+    call autopairs#Keybinds#SetEventignore()
+    normal a:cmd
+    call autopairs#Keybinds#ResetEventignore()
+    return ''
+endfunction
+
+function! autopairs#Keybinds#SetEventignore()
+    let g:autopairs_oldeventignore = &eventignore
+    set eventignore+=InsertEnter,InsertLeavePre,InsertLeave
```

Comment:
Sigh, never mind. [8.2.1874](https://github.com/vim/vim/commit/b53e13a91ab2fc9d52bd044715daa84972f4ce47) adds InsertLeavePre. What a trash error message. `autocmd` gives a sane error message at least.

Anyway, maybe something like this?

```suggestion
    set eventignore+=InsertEnter,InsertLeave
    if v:version == 802 && has('patch1874') || v:version > 802
        set eventignore+=InsertLeavePre
    endif
```

I have no idea how to do an equivalent check for Neovim; `v:version` in nvim compiled some time after 0.5.0 was released is `800`, making this approach useless. Though I think we can get away with leaving out the event, so if you don't know happen to know, we can just leave it at this. I believe `InsertEnter` and `InsertLeave` are the most critical events anyway, though we can just kick that can down the road.

Reply:
Okay I've removed `InsertLeavePre` for now